plaso/parsers/ips_parser.py

```diff
@@ -0,0 +1,165 @@
+      return None
+
+    header = re.search('{.+?}', raw_content)
+
+    if header:
+      try:
+        structured_header = json.loads(header.group())
+        return structured_header
+
+      except (json.decoder.JSONDecodeError, TypeError):
+        return None
+
+    return None
+
+  def Open(self, file_object):
+    """Opens an IPS log file.
+
+      Args:
+      file_object (dfvfs.FileIO): file-like object.
+
+      Raises:
+        ValueError: if the file object is missing.
+    """
+
+    if not file_object:
+      raise ValueError('Missing file object.')
+
+    raw_content = file_object.read()
+    if isinstance(raw_content, bytes):
+      try:
+        raw_content = raw_content.decode()
+      except ValueError:
+        pass
+
+    self.header = self.GetHeader(raw_content)
+    self.content = self.GetContent(raw_content)
+
+
+class IPSParser(interface.FileEntryParser):
+  """Parses IPS formatted log files."""
+
+  NAME = 'ips'
+  DATA_FORMAT = 'IPS log file'
+
+  _plugin_classes = {}
+
+  def _ParseFileEntryWithPlugin(
+      self, parser_mediator, plugin, ips_file, display_name):
+    """Parses an IPS log  file entry with a specific plugin.
+
+    Args:
+      parser_mediator (ParserMediator): parser mediator.
+      plugin (IPSPlugin): IPS parser plugin.
+      ips_file (IPSFile): IPS file.
+      display_name (str): display name.
+    """
+    required_keys_exists = plugin.CheckRequiredKeys(ips_file)
+
+    if not required_keys_exists:
+      logger.debug('Skipped parsing file: {0:s} with plugin: {1:s}'.format(
+          display_name, plugin.NAME))
+      return
+
+    logger.debug('Parsing file: {0:s} with plugin: {1:s}'.format(
+        display_name, plugin.NAME))
+
+    try:
+      plugin.UpdateChainAndProcess(parser_mediator, ips_file=ips_file)
+
+    except Exception as exception:  # pylint: disable=broad-except
+      parser_mediator.ProduceExtractionWarning((
+          'plugin: {0:s} unable to parse IPS file with error: '
+          '{1!s}').format(plugin.NAME, exception))
+
+  def ParseFileEntry(self, parser_mediator, file_entry):
+    """Parses an IPS formatted log file entry.
+
+    Args:
+      parser_mediator (ParserMediator): parser mediator.
+      file_entry (dfvfs.FileEntry): file entry to be parsed.
+    """
+    ips_file = IPSFile()
+    file_object = file_entry.GetFileObject()
+
+    display_name = parser_mediator.GetDisplayName()
+
+    try:
+      ips_file.Open(file_object)
+
+    except ValueError as exception:
+      parser_mediator.ProduceExtractionWarning(
+          '[{0:s}]unable to parse file: {1:s} with error: {2!s}'.format(
+              self.NAME, display_name, exception))
+      return
+
+    if ips_file.header and ips_file.content:
+      try:
+        for plugin in self._plugins:
+          self._ParseFileEntryWithPlugin(
+              parser_mediator, plugin, ips_file, display_name)
+
+      except Exception as exception:  # pylint: disable=broad-except
```

Comment:
@rick-slin why the double catch all Exception ?

Reply:
That is an oversight on my part. I can see that the one here is redundant with the one in _ParseFileEntryWithPlugin

Reply:
That is an oversight on my part. I can see that the one here is redundant with the one in _ParseFileEntryWithPlugin.